fieldsolver/gridGlue.cpp

```diff
@@ -179,6 +189,244 @@ void feedMomentsIntoFsGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
 
   MPI_Waitall(sendRequests.size(), sendRequests.data(), MPI_STATUSES_IGNORE);
 
+  // ---------------------------------------------------------------------------------
+  // Boxcar Smoothening
+  const int *mntDims= &momentsGrid.getLocalSize()[0];
+  std::array<Real,fsgrids::moments::N_MOMENTS> *Grid;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *swap;
+  FsGrid<std::array<Real, fsgrids::moments::N_MOMENTS>, 2> swapGrid = momentsGrid;
+
+  /*Kernel Weights are fractional numbers 
+
+              std::array weights[i]
+    weight= ---------------------------
+                2*halfStencilWidth+1
+
+  and are calculated as shown above while iterating over each dimensions
+  The numerator is calculated filling the weights array
+  and the denominator depends only on the stencil width. 
+  */
+  Real weightC, weightL1, weightL2, weightR1, weightR2;
+  int maxRefLevel = mpiGrid.mapping.get_maximum_refinement_level();
+  int minRefLevel = 0;
+  const int maxStencilWidth=5;
+  int blurPasses = 3;
+  int refLevel;
+  int halfStencilWidth;
+  int switchPoint=1;
```

Comment:
Declared a switchPoint for specifying the switch between 3 and 5 point stencils.